sklearn/preprocessing/_polynomial.py

```diff
@@ -250,19 +253,29 @@ def fit(self, X, y=None):
             "constant",
             "linear",
             "continue",
+            "periodic",
         ):
             raise ValueError(
                 "extrapolation must be one of 'error', "
-                "'constant', 'linear' or 'continue'."
+                "'constant', 'linear', 'continue' or 'periodic'."
             )
 
         if not isinstance(self.include_bias, (bool, np.bool_)):
             raise ValueError("include_bias must be bool.")
 
         # number of knots for base interval
         n_knots = base_knots.shape[0]
+
+        if self.extrapolation == "periodic" and n_knots <= self.degree:
+            raise ValueError("Periodic splines require degree < n_knots.")
```

Comment:
Please recall the effective values for `n_knots` and  and `degrees` in the error message.